### PR TITLE
bazel: enable experimental_guard_against_concurrent_changes

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -1,9 +1,10 @@
-startup --host_jvm_args=-Xmx8g
+startup --host_jvm_args=-Xmx6g
```
Comment (lance6716) [inline .bazelrc:1]: what's the reason of this change?

**Review by lance6716 — APPROVED** [review]

**Review by xhebox — APPROVED** [review]
